### PR TITLE
LibJS: Support BigInt number formatting with Intl.NumberFormat

### PR DESCRIPTION
**Pull request by trflynn89**
This has been a todo for a while now:
```
test/intl402/BigInt    11/11    (100.00%) [ ✅ 11 ] 
```

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp [hunk 1]
```diff
@@ -1245,8 +1313,17 @@ ThrowCompletionOr<void> set_number_format_unit_options(GlobalObject& global_obje
...
-Optional<Variant<StringView, String>> get_number_format_pattern(NumberFormat& number_format, double number, Unicode::NumberFormat& found_pattern)
+Optional<Variant<StringView, String>> get_number_format_pattern(NumberFormat& number_format, Value number, Unicode::NumberFormat& found_pattern)
 {
+    auto as_number = [&]() {
+        if (number.is_number())
+            return number.as_double();
+
+        // FIXME: This should be okay for now as our naive Unicode::select_pattern_with_plurality implementation
+        //        checks against just a few specific small values. But revisit this if precision becomes a concern.
+        return number.as_bigint().big_integer().to_double();
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp:1347]: Should this be a `VERIFY_NOT_REACHED()` in the first commit like all the others?

File: Userland/Libraries/LibJS/Tests/builtins/Intl/NumberFormat/NumberFormat.prototype.format.js [hunk 2]
```diff
@@ -1019,3 +1009,57 @@ describe("style=unit", () => {
...
+    test("default", () => {
+        const en = new Intl.NumberFormat("en");
+        expect(en.format(1n)).toBe("1");
+        expect(en.format(12n)).toBe("12");
+        expect(en.format(123n)).toBe("123");
+
+        const ar = new Intl.NumberFormat("ar");
+        expect(ar.format(1n)).toBe("\u0661");
+        expect(ar.format(12n)).toBe("\u0661\u0662");
+        expect(ar.format(123n)).toBe("\u0661\u0662\u0663");
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Intl/NumberFormat/NumberFormat.prototype.format.js:1023]: can we add a test for a truly *big* integer here?

File: Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp:1347]: Oops, yeah that was a bad squash

File: Userland/Libraries/LibJS/Tests/builtins/Intl/NumberFormat/NumberFormat.prototype.format.js (hunk 2 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Tests/builtins/Intl/NumberFormat/NumberFormat.prototype.format.js:1026]: Done!

**Review by linusg — APPROVED** [review]
Very nice :^)

**Pull request merged**